src/librados/IoCtxImpl.cc

```diff
@@ -2020,9 +2018,13 @@ void librados::IoCtxImpl::C_aio_Complete::finish(int r)
   c->cond.Signal();
 
   if (r == 0 && c->blp && c->blp->length() > 0) {
-    if (c->out_buf && !c->blp->is_provided_buffer(c->out_buf))
-      c->blp->copy(0, c->blp->length(), c->out_buf);
-    c->rval = c->blp->length();
+    if (c->blp == &c->bl && !c->blp->is_contiguous()) {
```

Comment:
This judgement based on the implementation of msg.  i think we should used code like other places to judge which by length. @tchaikov . What's your suggestion?

Reply:
@agutikov please note in 
```int librados::IoCtxImpl::aio_read(const object_t oid, AioCompletionImpl *c,
				  bufferlist *pbl, size_t len, uint64_t off,
				  uint64_t snapid, const blkin_trace_info *info)```
`c->blp` is assigned to `pbl`, and `c->bl` is not set. probably we still need `out_buf` as an indicator?